### PR TITLE
Move benchmarks to `mlp-gen`

### LINKED ISSUE #533: Use `mlp-gen` for benchmarks
We're benchmarking a lot of matmul/fully-connected kernels, but we generated the MLIR files with some scripts / by hand, when we already have an MLP generator.

We should replace all (MLP-related) MLIR files in `benchmarks/mlir` with calls to `mlp-gen` with the appropriate flags, piping the result to `tpp-run`.

@adam-smnk

### PR DESCRIPTION
**Pull request by adam-smnk**
Replaces matmul and fc generated benchmark files with runtime IR generator using `mlp-gen`.

* Adds a new benchmark type 'IR-GEN' to allow running MLIR IR created by any external generator.
* Extends benchmarks harness to accept STDIN input.
* Extends `mlp-gen` with matmul and fully connected kernel generators.
* Aligns `mlp-gen` packing with `pack-matmul`.
* Improves  `mlp-gen` bias and relu IR generation to avoid temporary buffers.
* Adds option to  `mlp-gen`  to apply VNNI packing at the time of IR generation.
* Adds metadata generation to `mlp-gen` to attach to kernels required 'RUN:' strings and total flops count.

Fixes #533

### CODE REVIEW

File: tools/mlp-gen/MLPGen.h [hunk 1]
```diff
@@ -18,6 +18,8 @@
 
 #include "TPP/BuilderUtils.h"
 
+#include <optional>
```
Comment (adam-smnk) [inline tools/mlp-gen/MLPGen.h:21]: Not needed anymore

File: tools/mlp-gen/MLPGen.cpp [hunk 2]
```diff
@@ -411,53 +534,66 @@ AffineMap MLPGenerator::getMap(Value tensor, MapType type) {
...
+      getDims({0, 2, 4, 6});
+    } else if (packed)
+      getDims({0, 2, 3, 5});
+    else
+      getDims({0, 2});
+    break;
+  case MAP_MATMUL_WEIGHT:
+    // Packed tensors have 4/5 dims and 6 loops (ppr-ppr)
+    n = packed ? 6 : 3;
+    // Extra VNNI packing reduction dim
```
Comment (adam-smnk) [inline tools/mlp-gen/MLPGen.cpp:574]: Remove duplicate

**Comment by rengolin** [comment]
PDF results look identical!

**Review by rengolin — APPROVED** [review]
Looks great, thanks!

**Pull request merged**